evm/src/cpu/kernel/interpreter.rs

```diff
@@ -0,0 +1,301 @@
+use ethereum_types::{U256, U512};
+
+struct Interpreter<'a> {
+    code: &'a [u8],
+    offset: usize,
+    stack: Vec<U256>,
+}
+
+pub fn run(code: &[u8], initial_offset: usize, initial_stack: Vec<U256>) -> Vec<U256> {
+    let mut interpreter = Interpreter {
+        code,
+        offset: initial_offset,
+        stack: initial_stack,
+    };
+    // Halt the execution if a jump to 0xdeadbeef was done.
```

Comment:
I think it would be nice if we handle `STOP` (`0x00`) instead, and then interpret out-of-range offsets as zero, to match the EVM behavior. The tests could still terminate by jumping to `0xdeadbeef`, since it should always be out of range (unless our kernel grows to ~4mb), resulting in `STOP`.

Reply:
Hm can you jump to an out-of-range location in the EVM? It would seem that this would panic since an out-of-range location can't be a JUMPDEST. 
But yes it's a good idea to support STOP. I thought about adding a global function that we can call in tests to stop execution and examine the stack
```
global stop:
  JUMPDEST
  STOP
```
but one issue is that we probably don't want this function in the production code (?) so we'd have to add it only for tests.